### PR TITLE
[REVIEW] Fix handling of empty inputs for concatenate

### PR DESCRIPTION
**Pull request by jrhemstad**
Closes https://github.com/rapidsai/cudf/issues/4629

- Throw if no columns were passed to concatenate
- Return empty column if all columns to concat are empty

### CODE REVIEW

File: cpp/src/copying/concatenate.cu [hunk 1]
```diff
@@ -354,13 +354,19 @@ concatenate(std::vector<column_view> const& columns_to_concat,
             rmm::mr::device_memory_resource *mr,
             cudaStream_t stream) {
 
-  if (columns_to_concat.empty()) { return std::make_unique<column>(); }
+  CUDF_EXPECTS(not columns_to_concat.empty(),
+               "Unexpected empty list of columns to concatenate.");
```
Comment (kkraus14) [inline cpp/src/copying/concatenate.cu]: If there's no columns to concat can we not just return an empty table? I could imagine from the python side:
```
cudf.concat([cudf.DataFrame(), cudf.DataFrame()])
```

Comment (jrhemstad) [reply cpp/src/copying/concatenate.cu]: This function returns a column. If there aren't any columns to concat, we don't know the type of the column to return.

Comment (jrhemstad) [reply cpp/src/copying/concatenate.cu]: If you try and concat zero tables you'll get a table with zero columns in return: https://github.com/rapidsai/cudf/blob/2f136f0893bee595853e3080afe0619e2e4bfc09/cpp/src/copying/concatenate.cu#L422

Comment (kkraus14) [reply cpp/src/copying/concatenate.cu]: Woops. Is this situation handled at a higher level in the `Table` concatenate API?

Comment (jrhemstad) [reply cpp/src/copying/concatenate.cu]: Yes, see above.

**Pull request merged**